tensorflow/python/training/experimental/mixed_precision.py

```diff
@@ -73,39 +73,96 @@ def _wrap_optimizer(opt, loss_scale):
-  and is used to make the model run faster. This function will use mixed
-  precision to speed up the execution time of your model when run on a GPU. It
-  does this by changing the dtype of certain operations in the graph from
-  float32 to float16.
-
-  This function additionally wraps an Optimizer with a LossScaleOptimizer, which
-  is required to prevent underflow in the float16 tensors during the backwards
-  pass. An optimizer must be passed to this function, which will then be wrapped
-  to use loss scaling.
+  """Enables faster training on compatible hardware accelerators by using a
+  mix of numerical data types. Known as mixed precision, this is achieved
+  via a graph rewrite operation and a scaled optimizer.
+  
+  Mixed precision is the use of both float16 and float32 tensors when training
+  a model. Performing only certain arithmetic operations in lower precision
+  (float16) results in performance gains of when using compatible hardware
+  accelerators. This decreases training time and reduces memory usage,
+  typically without sacrificing model performance.
+
+  Example:
+
+  ```python
+  model = tf.keras.models.Sequential([
+      ...
+  ])
+
+  opt = tf.keras.optimizers.SGD()
+  opt = tf.train.experimental.enable_mixed_precision_graph_rewrite(opt)
+
+  model.compile(loss="categorical_crossentropy",
+                optimizer=opt,
+                metrics=["accuracy"])
+
+  model.fit(x_train, y_train,
+            batch_size=batch_size,
+            epochs=epochs)
+  ```
+
+  For a complete example showing the speed-up on training an image classification
+  task on CIFAR10, check out this
+  [Colab notebook](https://colab.research.google.com/drive/1qabCQKJZhdIDOrXuhoLa9uMB4RJJ2a1N).
+
+  Calling `enable_mixed_precision_graph_rewrite(opt)` enables the graph rewrite
+  operation before computing gradients. The function additionally returns an
+  `Optimizer`(`opt`) wrapped with a `LossScaleOptimizer`. This prevents underflow
+  in the float16 tensors during the backward pass. An optimizer of type
+  `tf.train.Optimizer` or `tf.keras.optimizers.Optimizer` must be passed to this
+  function, which will then be wrapped to use loss scaling.
+
+  <img src="https://i.imgur.com/5kA9pcS.png" width="500px">
+
+  The graph rewrite operation changes the `dtype` of certain operations in the
+  graph from float32 to float16. There are several categories of operations that
+  are considered or rejected by this rewrite operation, and they are defined in
+  [auto_mixed_precision_lists.h](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h),
+  and each category is summarized below:
+
+  * [`ClearList`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L178): Ops that do not have numerically significant effects. E.g. `ArgMax` and `Floor`.
```

Comment:
The line numbers of these links will change over time as new ops are added. Maybe the text should explain that each corresponds to the function in that file.